messaging/broker.go

```diff
@@ -92,25 +126,67 @@ func (b *Broker) Open(path string, u *url.URL) error {
 	}
 	b.path = path
 
-	// Require a non-blank connection address.
-	if u == nil {
-		return ErrConnectionAddressRequired
+	// Ensure root directory exists.
+	if err := os.MkdirAll(path, 0700); err != nil {
+		b.close()
+		return fmt.Errorf("mkdir: %s", err)
+	}
+
+	// Open meta file.
+	meta, err := bolt.Open(b.metaPath(), 0600, &bolt.Options{Timeout: 1 * time.Second})
+	if err != nil {
+		b.close()
```

Comment:
I can get behind that. I don't love the extra indentation but I dislike missing `close()` statements more. :)

Added: b045ad5